sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -38,6 +39,13 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
     legacyFormat = FAST_DATE_FORMAT,
     isParsing = true)
 
+  private val timestampNTZFormatter = TimestampFormatter(
+    options.timestampFormatInRead,
```

Comment:
I think we should fail earlier if the pattern string contains timezone? It doesn't make sense to have timezone in NTZ string.

Reply:
moreover, maybe we should add a new CSV option to define pattern for timestamp ntz.

Reply:
Updated.